### ISSUE TITLE
No Idle Warrior Graphics for South West Direction With Certain Gear

### ISSUE THREAD

**Issue by NiteKat** [diasurgical/devilutionX#3071]
**Important information**
Windows 10 / DevilutionX version / Compiled myself

**Describe**

**To Reproduce**
1. Equip a warrior with a blunt weapon, shield, and medium armor.
2. Enter the dungeon.
3. Move down and to the left one tile (you probably spawn facing the bad direction, but this just confirms it)
4. Witness your character stand facing west instead of south west.

**Expected behavior**
In the top screenshot below, Warrior should be facing the same direction as the bottom screenshot (only difference between them was unequipping the shield)

**Screenshots**
![image](https://user-images.githubusercontent.com/28605404/136663505-36af8cf9-97ec-46ff-9ee4-cc0b55b13586.png)
![image](https://user-images.githubusercontent.com/28605404/136663604-1ea87340-bf81-4319-a863-49db4d9bf823.png)

**Additional context**
This problem is also present in Vanilla Hellfire. Unsure if it is present in Vanilla 1.09 (I know there are many sprite issues in 1.09, but unsure of this one specifically).

**Comment by AJenbo** [comment]
The sprite does appear to be in the data: 
![image](https://user-images.githubusercontent.com/204594/136664182-5f115396-5d1c-43d2-9cc4-18adcfa2ece4.png)

**Comment by julealgon** [comment]
There are a few other instances of similar problems, I wonder if it would be best for us to just group them into a single issue, or at least link them together.

I know there is a problem with the rogue sprite in one of the directions (using staff maybe)? And others as well.

**Comment by FitzRoyX** [comment]
Couldn't we just edit the sword pixels out of the sword sprites and put it in the devx mpq?

Before you do that, though, why am I seeing this direction in the spritesheet here?

**Comment by AJenbo** [comment]
Yeah, masking would probably work. We might implement the needed bits for 1.3.0, but there most likely won't be time to fix this sprite in time.

**Comment by Chonkblonk** [comment]
While my observation is linked here, I noticed that with the Rogue/Bard wearing Medium armors, if she's holding a sword it will show as a straight blade when she's standing idle, or attacking, but when she's walking the straight blade becomes a curved scimitar-like one.

This would be a pretty easy error to rectify by just manually editing the sprites (since she retains the same pose in Light and Medium armor, and could be just copied from there), but that's not really an option to include with DevilutionX, and I don't know if any kind of masking approach would be viable.

A thought that strikes me is that graphical fixes like these could be made (where dual wielding frames for the Bard could also be included), and that DevilutionX does not PROVIDE these frames, but rather that the frames could be automatically loaded by DevilutionX if the user provides an .MPQ file with them to the program.
Similar to how for Doom, there exists the so called "Sprite Fixing Project", which cleans up stuff like stray pixels and fixes sprite offsets for the monsters in that game. Those are made from the commercial assets, and while all the sourceports (and even the original .exe) are compatible with them, none of the ports actually include those, so it's not any potential legal or licensing issue for them.

Thus I'm thinking, a Diablo Sprite Fixing Project could theoretically exist, which stones rough edges like these.
iD Software has always had a very laid back attitude to modding (including for a lot of transformative stuff), but I'm not sure if Blizzard has the same chill.

**Comment by StephenCWills** [comment]
> I don't see the difference, you'd still be hosting copyrighted content, wouldn't you, just in a different repository?

The difference between what? Assuming you're considering hosting the content here, that isn't an option in the first place because we're not going to allow it. So it would have to go in a different repository, presumably owned/maintained by someone who's willing to take that risk.
